### DISCUSSION TITLE
how to use SimulateForwardMagneticField module

### DISCUSSION THREAD

**Discussion by zhifeiji-han** [SCIInstitute/SCIRun#2316] [General]
I want to use  SimulateForwardMagneticField module, and this module has four input Interface and two output Interface. I want to know every interface meaning and  method of use, maybe a simple example of this module is helpful. When I click help of the module, I get a simple explaine, but still don't know how to use this module.
![image](https://user-images.githubusercontent.com/70640243/118801993-35cae900-b8d4-11eb-83b6-07734be1c567.png)

I'd like to know if there is a file or website for specific module usage, and if there is a simple example program for each module.

**Comment by dcwhite** [comment]
@moritzdannhauer

**Comment by dcwhite** [comment]
This package (https://github.com/SCIInstitute/BrainStimulator) has a network that uses this module, but I don't know how much additional documentation is there.

**Comment by zhifeiji-han** [comment]
I read the network（TMS_TOY_EXAMPLE.srn5）, it's a bit complicated, I don't understand how to use SimulateForwardMagneticField module. Can you explain how to use this module in this example？
![image](https://user-images.githubusercontent.com/70640243/120147181-d8f6f900-c218-11eb-8449-d358f7c29688.png)

**Comment by moritzdannhauer** [comment]
Hi zhifeiji-han,
The module 'SimulateForwardMagneticField' is simple (source code) but a bit particular.  
From the 4 inputs, I think the idea was to use either electric field (1) + conductivity tensor (2) or current density right away as dipolar current (3) to compute mag. field at detector locations (4). The two outputs give you the mag. field as vector (1) or magnitude (vector norm, (2)) at the detector locations. It requires all 4 inputs even though I remember to input the relevant information through (3) and (4).
I attached a very(!) simple example for SCIRun 4 and 5 and quickly typed in some MATLAB code to compare results. I do not know what you are trying to do but I hope this helps a little.
Best,
Moritz

[test_scirun_simmag.zip](https://github.com/SCIInstitute/SCIRun/files/6582227/test_scirun_simmag.zip)

**Comment by dcwhite** [comment]
Thanks a lot Moritz!

**Comment by zhifeiji-han** [comment]
Thanks Moritz, I understand this module.

**Comment by zhifeiji-han** [comment]
For a module with multiple input and output ports, a simple application example is given, which is very helpful for learning and using the module.
